Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -95,5 +166,180 @@ public static class CustomEntitySearch
+                                });
+                        entitiesFound.Add(checkMatch);
+                    }
+                }
+
+
+            }
+        }
+
+        private static double DamerauLevenshteinCalculation(string text, string checkMatch)
+        {
+            double[,] dynamicDistanceCalc = new double[text.Length + 1, checkMatch.Length + 1];
+            double substitutionCost = -1;
+            double accentAddition = 1;
+            for (int currTextIndex = 0; currTextIndex <= text.Length; currTextIndex++)
+                dynamicDistanceCalc[currTextIndex, 0] = currTextIndex;
+            for (int currWordIndex = 0; currWordIndex <= checkMatch.Length; currWordIndex++)
+                dynamicDistanceCalc[0, currWordIndex] = currWordIndex;
+
+            for (int currTextIndex = 0; currTextIndex < text.Length; currTextIndex++)
+            {
+                for (int currWordIndex = 0; currWordIndex < checkMatch.Length; currWordIndex++)
+                {
+                    if (text[currTextIndex].Equals(checkMatch[currWordIndex]))
+                        substitutionCost = 0;
+                    else if (String.Compare(checkMatch[currWordIndex].ToString(), text[currTextIndex].ToString(),
+                                    CultureInfo.CurrentCulture, CompareOptions.IgnoreNonSpace) == 0)
+                        substitutionCost = .5;
+                    else
+                        substitutionCost = 1;
+                    if (checkMatch[currWordIndex].IsAccent() ^ text[currTextIndex].IsAccent())
+                        accentAddition = 0.5;
+                    dynamicDistanceCalc[currTextIndex + 1, currWordIndex + 1] = Math.Min(
+                        Math.Min(dynamicDistanceCalc[currTextIndex, currWordIndex + 1] + accentAddition, // deletion
+                        dynamicDistanceCalc[currTextIndex + 1, currWordIndex] + accentAddition), // insertion
+                        dynamicDistanceCalc[currTextIndex, currWordIndex] + substitutionCost); // substitution
+                    if (currTextIndex > 0 && currWordIndex > 0 && text[currTextIndex].Equals(checkMatch[currWordIndex - 1]) && checkMatch[currWordIndex].Equals(text[currTextIndex - 1]))
+                        dynamicDistanceCalc[currTextIndex + 1, currWordIndex + 1] = Math.Min(dynamicDistanceCalc[currTextIndex + 1, currWordIndex + 1], 
+                            dynamicDistanceCalc[currTextIndex - 1, currWordIndex - 1] + substitutionCost); // transposition
+                }
+            }
+
+            return dynamicDistanceCalc[text.Length, checkMatch.Length];
+        }
+
+        /*
+        * Given an entity the user wants to find, this method removes delineating characters if they are found in the
+        * beginning or end of the entity definition. The method then returns the exact word that will be used for fuzzy matching
+        */
+        public static string CreateWordArray(string checkMatch)
```

Comment:
>CreateWordArray [](start = 29, length = 15)

nit: rename to maybe "TrimDelineatingCharacters()" since you use it for text as well as words